crates/metal-codegen-llvm/src/core.rs

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+//! Compilation step for modules from MIR to LLVM IR
+
+use std::ffi::CString;
+
+use llvm_sys::{bit_writer, core::LLVMPrintModuleToString};
+use metal_mir::parcel::Module;
+
+use crate::{safeties::MemoryBuffer, CodeGenValue, LLVMRefs};
+
+/// Compiles an LLVM module and returns either human-readable
+/// LLVM IR or LLVM bytecode depending on `human_readable`.
+#[inline]
```

Comment:
Why?

Reply:
> Why?

https://nnethercote.github.io/perf-book/inlining.html

Reply:
> > Why?
> 
> [nnethercote.github.io/perf-book/inlining.html](https://nnethercote.github.io/perf-book/inlining.html)

https://m.xkcd.com/1691/

Reply:
> > > Why?
> > 
> > 
> > [nnethercote.github.io/perf-book/inlining.html](https://nnethercote.github.io/perf-book/inlining.html)
> 
> https://m.xkcd.com/1691/

it's a single line smh

Reply:
Sorry?
![изображение](https://github.com/user-attachments/assets/15977aed-7a62-446b-a7df-98ab389c7a29)

Reply:
> Sorry? ![изображение](https://private-user-images.githubusercontent.com/77104725/390406031-15977aed-7a62-446b-a7df-98ab389c7a29.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MTA3MDQsIm5iZiI6MTczMjcxMDQwNCwicGF0aCI6Ii83NzEwNDcyNS8zOTA0MDYwMzEtMTU5NzdhZWQtN2E2Mi00NDZiLWE3ZGYtOThhYjM4OWM3YTI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI3VDEyMjY0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0MDM0ZTc0YTM2OTg5ODAzZmYwYWQyMDM4YWEzNWRlZmIyMmU4MGYzOTgzNWQ5ZDdjOGQzYjczYmY3YTUzYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.b7boyvHDa4i-L-S4ge5Ljsf9Qgq4Tzuj2i1ZEUOgQSU)

I meant the `#[inline]`

Reply:
Inline affects the whole function?

Reply:
> Inline affects the whole function?

inlining would technically not make the function a function, but
just replace the call with the function body (at least from my understanding.)

Which should be fine/better for us since `compile_module` at most is gonna be used in
one or two places, both in `core` here.

Reply:
Inline is not used because it's "fine", it's a tool for improving performance (and sometimes allowing debugging with `inline(never)`). Only and *only* if it has been confirmed that inlining the function indeed improves performance should `#[inline]` (and god forbid `#[inline(always)]` be added. As far as my intuition goes, none of this code was ever executed, not speaking of measuring it's performance (would love to be proven wrong though). That's why i linked the XKCD comic, this `#[inline]` here is a clear case of premature optimization